monadic-rx/shared/src/main/scala/mhtml/rx.scala

```diff
@@ -165,11 +166,16 @@ object Rx {
   final case class Zip     [A, B]     (self: Rx[A], other: Rx[B])                   extends Rx[(A, B)]
   final case class DropRep [A]        (self: Rx[A])                                 extends Rx[A]
   final case class Merge   [A, B >: A](self: Rx[A], other: Rx[B])                   extends Rx[B]
+  final case class MergeN  [A, B >: A](self: List[Rx[A]])                           extends Rx[B]
   final case class Foldp   [A, B]     (self: Rx[A], seed: B, step: (B, A) => B)     extends Rx[B]
   final case class Collect [A, B]     (self: Rx[A], f: PartialFunction[A, B], b: B) extends Rx[B]
   final case class SampleOn[A, B]     (self: Rx[A], other: Rx[B])                   extends Rx[A]
   final case class Imitate [A]        (self: Var[A], other: Rx[A])                  extends Rx[A]
 
+  implicit class RichListRx[A](val value: List[Rx[A]]) extends AnyVal {
```

Comment:
Not sure about adding stuff to `List`... I would simply add that to the other methods with a varargs, it's would also be better for discoverability:

```scala
rx.mergeN(rxb, rxc, rxd)
```

Reply:
Having it on list feels more natural, once you know about it, and it should only work on `List[Rx[...]]` - at least, IntelliJ seems smart enough to get the completion correct only in the valid case (for what it's worth). We could also provide both interfaces.

Reply:
Your version would fail at runtime on `List.empty[Rx[Uni]].merge`. Also, the clean way to add methods on standard types would be do to it under an `mhtml.syntax._` import, which definitely feels overkill for a single method...

Another way would be `Rx.mergeN(rxa, rxb, rxb)`.

Reply:
Hmm, this did not generate any runtime errors for me, but maybe I misunderstood:

```scala
  test("RichListRx") {
    val x = List.empty[Rx[Unit]].merge
  }
```


There has to be a first method ;) More seriously though, I suspect there will be others to follow as we flesh out what we want to do on lists of Rx in practice.

I do like the `Rx.mergeN` option better than `rxa.mergeN(rxb, rxc, rxd)`, but would still be nice if the optional `mhtml.syntax` import as you suggest. I guess it is entirely possible for the user to create their own `RichListRx` since `MergeN` is not `private`, but I don't see the harm in supplying a few for convenience and examples; we could even potentially have alternatives for different collections (`List`, `Seq`, `Array`, ..) that might be too much, just an idea.

Reply:
```scala
val x = List.empty[Rx[Unit]].merge
```

This is an empty Rx, which breaks the fundamental non empty assumption of every Rx in mhtml :P 

Let's keep skip the syntax for now, a simple API like this should do the trick:

```scala
object Rx {
  def mergeN[Rx[A]](head: Rx[A], tail: Rx[A]*): Rx[A]
}
```

Reply:
Oh, I see, that would break things.

Here's another solution I think: 
```scala
    case MergeN(self) =>
      val selfRxs: List[Rx[A]] = self.headOption match {
        case Some(_) => self
        case None => List(Rx(null.asInstanceOf[A]))
      }
      val ccs: List[Cancelable] = selfRxs.map(rx => run(rx)(effect))
      Cancelable { () => ccs.foreach(cc => cc.cancel) }
```

Some minimal runtime overhead there for the check, but I guess there would also be some runtime overhead every time a user has to remove an element from the head of a list, the prepend it back (more or less depending on what collection we are using), with the additional typing/screenspace overhead of having to do so each time.